src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -688,36 +696,55 @@ void MemoryInput::initOptimalPrimitiveDescriptor() {
 
 // @todo add ascii diagramm for memory mapping / reuse
 void MemoryInput::createPrimitive() {
-    MemoryInputBase::createPrimitive();
     if (haveSubgraph()) {
-        OPENVINO_ASSERT(getOriginalInputsNumber() == subGraph->inputsNumber(),
-                        "Number of node inputs must be equal the number of inner graph's inputs: ",
-                        getOriginalInputsNumber(),
-                        " != ",
-                        subGraph->inputsNumber());
-
-        std::vector<MemoryPtr> inputMemory;
-        for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
+        CPU_NODE_ASSERT(getParentEdges().size() == subGraph->inputsNumber(),
+                        "Number of node inputs must be equal the number of inner graph's inputs");
+
+        for (size_t i = 0; i < getParentEdges().size(); i++) {
             auto srcEdgeMem = getSrcMemoryAtPort(i);
             // create a separate input memory objects instead of share them. avoid data corruption.
             auto mem = std::make_shared<Memory>(getEngine(), srcEdgeMem->getDescPtr(), srcEdgeMem->getMemoryBlock());
             subgraphMemoryPtrs.push_back(mem);
-            inputMemory.emplace_back(std::move(mem));
         }
 
```

Comment:
In fact this is the same action that is performed in LoRA node. Could you please align the implementation?
```cpp
    // Workaround to avoid making LoRa node always executable (isExecutable() = true)
    // This way we update subgraph's input memory without performing an actual Infer() call
    for (size_t i = 0; i < getOriginalInputsNumber(); i++) {
        auto subgraphInputNode = m_graph.getInputNodeByIndex(i);
        auto subgraphInputMemory = subgraphInputNode->getDstMemoryAtPort(0);
        subgraphMemoryPtrs.emplace_back(subgraphInputMemory);
    }
```